File: system_api/systemHandler.py
import platform, subprocess

def shut_down():
    current_os = platform.system()
    if current_os == "Windows":
        cmd = "shutdown /s /t 1"
    if current_os == "Linux":
        cmd = "sudo shutdown -h now"
    if current_os == "Darwin":
        cmd = "sudo shutdown -h now"
    else:
        raise NotImplementedError("Unsupported platform")
    subprocess.call(cmd, shell=True)
def restart_machine():
    current_os = platform.system()
    if current_os ==  "Windows":
        cmd = "shutdown /r /t 0"
    if current_os ==  "Linux":
        cmd = "sudo reboot"
    if current_os ==  "Darwin":
        cmd = "sudo reboot"
    else:
        raise NotImplementedError("Unsupported platform")
    subprocess.call(cmd, shell=True)

File: system_api/system_handler.go
package main

import (
	"fmt"
	"os/exec"
	"runtime"
)

func ShutDown() error {
	platform := runtime.GOOS
	switch platform {
	case "windows":
		_ = exec.Command("shutdown", "/s", "/t", "1")
	case "linux", "darwin":
		_ = exec.Command("shutdown", "-h", "now")
	default:
		return fmt.Errorf("Unsupported platform")
	}
	return nil
}

func Restart() error {
	platform := runtime.GOOS
	switch platform {
	case "windows":
		_ = exec.Command("shutdown", "/r", "/t", "0")
	case "linux", "darwin":
		_ = exec.Command("sudo", "reboot")
	default:
		return fmt.Errorf("Unsupported platform")
	}
	return nil
}
